SurveyApp.DAL/Repositories/Repository.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Data.Entity;
+using System.Linq.Expressions;
+using SurveyApp.DAL.DataContext;
+using SurveyApp.DAL.Repositories.Interfaces;
+
+namespace SurveyApp.DAL.Repositories
+{
+    public class Repository<TEntity> : IRepository<TEntity> where TEntity : class
+    {
+        internal ApplicationContext context;
+        internal DbSet<TEntity> dbSet;
+
+        public Repository(ApplicationContext context)
+        {
+            this.context = context;
+            this.dbSet = context.Set<TEntity>();
+        }
+
+        public void Insert(TEntity entityToInsert)
+        {
+            dbSet.Add(entityToInsert);
+        }
+
+        public virtual IEnumerable<TEntity> Get(
```

Comment:
The return type should be IQueryable.  It will allow to implement paging without returning of all records from a table.